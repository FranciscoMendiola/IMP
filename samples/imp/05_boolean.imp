if true and not false then
  skip
else
  x := 0
