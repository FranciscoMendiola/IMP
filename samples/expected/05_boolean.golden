Token {tokenKind = KIf, lexeme = "if", position = 0}
Token {tokenKind = KTrue, lexeme = "true", position = 3}
Token {tokenKind = KAnd, lexeme = "and", position = 8}
Token {tokenKind = KNot, lexeme = "not", position = 12}
Token {tokenKind = KFalse, lexeme = "false", position = 16}
Token {tokenKind = KThen, lexeme = "then", position = 22}
Token {tokenKind = KSkip, lexeme = "skip", position = 29}
Token {tokenKind = KElse, lexeme = "else", position = 34}
Token {tokenKind = Id, lexeme = "x", position = 41}
Token {tokenKind = Assign, lexeme = ":=", position = 43}
Token {tokenKind = IntLit, lexeme = "0", position = 46}
